lib/FusionInventory/Agent/Task/Inventory/Win32/AntiVirus.pm

```diff
@@ -69,38 +70,43 @@ sub doInventory {
 }
 
 sub _getMcAfeeInfo {
-    my ($logger) = @_;
-
-    my $path;
-    if (is64bit() && defined getRegistryKey(path => 'HKEY_LOCAL_MACHINE/SOFTWARE/Wow6432Node/McAfee/AVEngine')) {
-        $path = 'HKEY_LOCAL_MACHINE/SOFTWARE/Wow6432Node/McAfee/AVEngine';
-    } else {
-        $path = 'HKEY_LOCAL_MACHINE/SOFTWARE/McAfee/AVEngine';
-    }
 
     my %properties = (
         DATFILEVERSION  => [ 'AVDatVersion',         'AVDatVersionMinor' ],
         ENGINEVERSION32 => [ 'EngineVersion32Major', 'EngineVersion32Minor' ],
         ENGINEVERSION64 => [ 'EngineVersionMajor',   'EngineVersionMinor' ],
     );
 
-    my $info;
+    my $regvalues = [ 'AVDatDate', map { @{$_} } values(%properties) ];
+
+    my ($info, $macafeeReg);
+
+    if (is64bit()) {
+        $macafeeReg = getRegistryKey(
+            path => 'HKEY_LOCAL_MACHINE/SOFTWARE/Wow6432Node/McAfee/AVEngine',
+        );
+    }
+
+    if (!$macafeeReg) {
+        $macafeeReg = getRegistryKey(
+            path => 'HKEY_LOCAL_MACHINE/SOFTWARE/McAfee/AVEngine',
+        );
+    }
+
+    return unless $macafeeReg;
```

Comment:
this is a information strategy change (check first path first, fallback on second), not a change in the way the registry is accessed.

Reply:
Yes, also, before the PR, the first getRegistryKey() call was just used to define the path and make getRegistryValue() calls later. Now we directly get a hash and this is now a little bit more simple and efficient, at least in my though.
I'll add a comment for that unless you think something can goes wrong that way.